sycl/source/detail/event_impl.cpp

```diff
@@ -217,8 +217,7 @@ void event_impl::wait_and_throw(
 
```

Comment:
Not related to this patch, but it seems this function is not thread safe. We accessing `Cmd` here which AFAIK can be removed in graph cleanup.

Reply:
I wrapped wait_and_throw under mutex